### PR TITLE
build: add support for dynamic file names in all npm scripts

### PR DESCRIPTION
**Pull request by siguse**
Add possibillity to give a name as argument to all the npm run scripts. If no name given the target of the script will be src/app.ts. Also the vim environment is started using the given file name.

### CODE REVIEW

File: scripts/filename.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const fs = require('fs');
+
+function fileName(name, checkAvail = true) {
+    if (name !== undefined) {
```
Comment (erhise) [inline scripts/filename.js:4]: do you use tabstop=4?

File: scripts/filename.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        const fileName = `src/${name}.ts`;
+        if (!checkAvail) {
```
Comment (erhise) [inline scripts/filename.js:6]: explicit boolean check, e.g. `checkAvail === false`?

File: scripts/run-vim.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const { fileName } = require('./filename');
+const editor = process.argv[2];
+const path = fileName(process.argv[3]);
+
+const child_process = require('child_process')
+
+var child = child_process.spawn(editor, ['-s', 'scripts/vim-sandbox.vim', `${path}`], {
+        stdio: 'inherit'
```
Comment (erhise) [inline scripts/run-vim.js:8]: `shiftwidth=7000`?

File: scripts/run-vim.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const { fileName } = require('./filename');
+const editor = process.argv[2];
+const path = fileName(process.argv[3]);
+
+const child_process = require('child_process')
+
+var child = child_process.spawn(editor, ['-s', 'scripts/vim-sandbox.vim', `${path}`], {
```
Comment (erhise) [inline scripts/run-vim.js:7]: Cool use of child_process!

**Review by erhise — APPROVED** [review]
:shipit:  looks brilliant!

**Comment by erhise** [comment]
@siguse feel free to merge at your liking

File: scripts/filename.js (hunk 2 above)
Comment (siguse) [reply scripts/filename.js:6]: Sure, we can do that.

File: scripts/filename.js (hunk 1 above)
Comment (siguse) [reply scripts/filename.js:4]: Yes, I do. Is this a tabstop=2 project? :-)

**Comment by siguse** [comment]
@erhise will go over the code style according to your comments first  👍

File: scripts/filename.js (hunk 1 above)
Comment (erhise) [reply scripts/filename.js]: Do your prefer 4 we can make it a mixed project with different tabstop all over the place ;)

**Pull request merged**